test/Microsoft.AspNetCore.Razor.Runtime.Test/Runtime/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -19,6 +19,138 @@ public class TagHelperDescriptorFactoryTest
 
         protected static readonly string AssemblyName = TagHelperDescriptorFactoryTestAssembly.Name;
 
+        public static TheoryData RequiredAttributeParserErrorData
+        {
+            get
+            {
+                Func<string, RazorError> error = (message) => new RazorError(message, SourceLocation.Zero, 0);
+
+                return new TheoryData<string, RazorError>
+                {
+                    { "name,", error(Resources.TagHelperDescriptorFactory_UnexpectedEndOfRequiredAttribute) },
+                    { " ", error(Resources.FormatHtmlTargetElementAttribute_NameCannotBeNullOrWhitespace("Attribute")) },
+                    { "n@me", error(Resources.FormatHtmlTargetElementAttribute_InvalidName("attribute", "n@me", '@')) },
+                    { "name extra", error(Resources.FormatTagHelperDescriptorFactory_InvalidRequiredAttributeCharacter('e')) },
+                    {
+                        "[name='unended]",
+                        error(Resources.FormatTagHelperDescriptorFactory_InvalidRequiredAttributeCharacterExpectedOther(']', '\''))
+                    },
+                    { "[name", error(Resources.TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace) },
+                    { "[ ]", error(Resources.FormatHtmlTargetElementAttribute_NameCannotBeNullOrWhitespace("Attribute")) },
+                    { "[n@me]", error(Resources.FormatHtmlTargetElementAttribute_InvalidName("attribute", "n@me", '@')) },
+                    { "[name='value'", error(Resources.TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace) },
+                    { "[name ", error(Resources.TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace) },
+                    { "[name=value ", error(Resources.TagHelperDescriptorFactory_CouldNotFindMatchingEndBrace) },
```

Comment:
Using the resources makes it hard to see the results of the test. Suggest `$"this message is the {expected} one"` to make these more understandable.